File: SDKs/github.com/ghes-2.18/1.1.4/python/src/sdk/models/operations/repos_list_languages.py
from dataclasses import dataclass, field
from typing import Optional


@dataclass
class ReposListLanguagesPathParams:
    owner: str = field(metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    repo: str = field(metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    

@dataclass
class ReposListLanguagesRequest:
    path_params: ReposListLanguagesPathParams = field()
    

@dataclass
class ReposListLanguagesResponse:
    content_type: str = field()
    status_code: int = field()
    language: Optional[dict[str, int]] = field(default=None)
    

File: SDKs/github.com/ghes-2.21/1.1.4/typescript/src/sdk/models/operations/reposlistlanguages.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class ReposListLanguagesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=owner" })
  owner: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=repo" })
  repo: string;
}


export class ReposListLanguagesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: ReposListLanguagesPathParams;
}


export class ReposListLanguagesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  language?: Map<string, number>;
}
